### ISSUE TITLE
Rendering performance regression between 0.15 rc2 and rc3 on MacOS

### ISSUE THREAD

**Issue by rudderbucky** [bevyengine/bevy#16315]
## Bevy version

* Original: 0.15.0-rc.2
* Current: 0.15.0-rc.3

## Relevant system information

Please include:

Rust: cargo 1.84.0-nightly (031049782 2024-11-01)
OS: macOS sequoia

## What's performing poorly?

FPS regression -> 120fps-90fps in my game.
Can be seen upon loading into the game, unfortunately source is not available.

## Before and After Traces

See zip 
[archive.zip](https://github.com/user-attachments/files/17689508/archive.zip)
[archive-novsync.zip](https://github.com/user-attachments/files/17689531/archive-novsync.zip)

**Comment by DGriffin91** [comment]
7950x, RTX3060
```
rc2    | rc3
 2.5ms |  2.8ms many_foxes
 2.7ms |  2.7ms many_cubes
26.3ms | 22.0ms many_cubes --vary-material-data-per-instance
17.4ms | 22.2ms many_buttons
 1.4ms |  1.5ms many_animated_sprites
```

**Comment by rudderbucky** [comment]
This is wgpu specific. It appears the `distance` function has a regression for fragment shaders running over large textures (~4000x4000).

**Comment by rudderbucky** [comment]
It's also metal-only.

**Comment by rudderbucky** [comment]
It's also M1 only... not seeing it on an M2 MBP.

**Comment by rudderbucky** [comment]
This is https://github.com/gfx-rs/wgpu/issues/6518

**Comment by rudderbucky** [comment]
@alice-i-cecile yes we do. wgpu has not yet pushed an option to disable the UB workaround that causes the regression. Once they do, we need to update to that version and figure out where to disable it which will be another discussion bevy needs to have. I think disabling it for non wasm is sufficient but others may not share that sentiment.

**Comment by alice-i-cecile** [comment]
Ugh, okay. I don't think we can block a release on this indefinitely: cutting from 0.15 milestone :(

> I think disabling it for non wasm is sufficient but others may not share that sentiment.

This is my stance too, at least for default behavior. We should probably allow it to be configurable regardless of platform.

**Comment by JMS55** [comment]
This should be fixed since https://github.com/bevyengine/bevy/pull/17767, probably.

**Comment by rudderbucky** [comment]
Thanks a bunch for pushing the fix to this @JMS55!

**Comment by JMS55** [comment]
No problem.
